src/storage/garbage_collector.cpp

```diff
@@ -13,25 +17,43 @@
 namespace terrier::storage {
 
 std::pair<uint32_t, uint32_t> GarbageCollector::PerformGarbageCollection() {
+  // Create the UndoBuffer for this GC run
+  delta_record_compaction_buffer_ = new UndoBuffer(txn_manager_->buffer_pool_);
+  // The compaction buffer is empty
+  compaction_buffer_empty_ = true;
```

Comment:
I don't think you need a separate boolean to track this, right? You can directly ask the buffer if it is empty.